geometry/line_segment_intersection.cpp

```diff
@@ -0,0 +1,88 @@
+#include <iostream>
+
+struct Point {
+    int x, y;
+};
+
+struct SegmentIntersection {
+    inline bool intersect(Point first_point, Point second_point,
+                                    Point third_point, Point forth_point) {
+       /**
```

Comment:
comment blocks should go above the function name i.e. for example, the comment block between lines 10-14 should be just above line 8.